packages/block-editor/src/components/block-list/block-wrapper.js

```diff
@@ -265,6 +268,23 @@ export function useBlockWrapperProps( props = {}, { __unstableIsHtml } = {} ) {
 		};
 	}, [ isNavigationMode, isHovered, setHovered ] );
 
+	useEffect( () => {
+		function onMouseEnter() {
+			addHoveredBlock( clientId );
+		}
+		function onMouseLeave() {
+			removeHoveredBlock( clientId );
+		}
+
+		ref.current.addEventListener( 'mouseenter', onMouseEnter );
+		ref.current.addEventListener( 'mouseleave', onMouseLeave );
```

Comment:
@youknowriad - RE (https://github.com/WordPress/gutenberg/issues/25469#issuecomment-695990845):
> About the block inserter disruption, I wonder if this is related to the use of React events instead of DOM events (event bubbling is distrupted), so I'd consider trying event handlers using the "ref" in the BlockWrapper component.

I think this is what you were proposing?  It still seems to have the same issue, as the `mouseleave` fires for a block when we hover over an inserter inside it.

One way to get around this would be to replace add/removeHoveredBlock with a `setHoveredBlocks` in both cases.  We could use `elementsFromPoint` to get the elements, make a list of the clientIds listed on those elements, and pass that list to `setHoveredBlocks`. 🤔

Reply:
I see. I don't have a specific idea, worth trying yours. One important thing to check is the performance impact for long posts when you're moving the mouse in the editor.

Reply:
Updated to try what I was thinking. It seems to behave well.